### PR TITLE
Feat: Data Download for new data structure

### PR DESCRIPTION
**Pull request by emre2038**
## Description
making data download work with the new data structure

## How to test it locally
* expand the rich data panel
* using the popup menu of any chart download data as CSV, Excel and JSON and confirm that the downloaded data is correct.

## Screenshots
image

### Updated
* `chart.js` to add / modify the functions that will be used to download data

## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design matches the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [ ]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/profile/chart.js [hunk 1]
```diff
@@ -407,4 +407,87 @@ export class Chart extends Observable {
         this.vegaView.signal('applyFilter', false).run()
       }
   };
+
+  getExportData = (isArray = false) => {
+    const data = this.vegaView.data('data_formatted');
+    const primaryGroup = this.vegaView._signals['mainGroup'].value;
```
Comment (milafrerichs) [inline src/js/profile/chart.js:413]: you should get the signals this way: 

```suggestion
    const primaryGroup = this.vegaView.signal("mainGroup")
```

File: src/js/profile/chart.js [hunk 2]
```diff
@@ -407,4 +407,87 @@ export class Chart extends Observable {
...
+      var downloadAnchorNode = document.createElement('a');
+      downloadAnchorNode.setAttribute("href", dataStr);
+      downloadAnchorNode.setAttribute("download", this.title + ".json");
+      document.body.appendChild(downloadAnchorNode); // required for firefox
+      downloadAnchorNode.click();
+      downloadAnchorNode.remove();
+  }
+
+  exportAsKml = () => {
+      const exportData = getExportData();
```
Comment (milafrerichs) [inline src/js/profile/chart.js:483]: you can remove the code for the KML export. it is not correct.

**Review by milafrerichs — CHANGES_REQUESTED** [review]

File: src/js/profile/chart.js [hunk 3]
```diff
@@ -1,264 +1,486 @@
...
+        this.vegaView.signal('applyFilter', false).run()
+      }
+  };
+
+  getExportData = (isArray = false) => {
+    const data = this.vegaView.data('data_formatted');
+    const primaryGroup = this.vegaView.signal('mainGroup');
+    const formatting = this.vegaView.signal('numberFormat');
+    const datatypes  = this.vegaView.signal('datatype');
+    let rows = [];
```
Comment (milafrerichs) [inline src/js/profile/chart.js:420]: I think you could replace the whole function with `papaparse`

We don't need to and probably should not format the data anyway. 
we just need to add the raw data to the download.

**Review by milafrerichs — CHANGES_REQUESTED** [review]

**Review by milafrerichs — APPROVED** [review]
